request/builder/builder.go

```diff
@@ -279,8 +281,18 @@ func (qb *Builder) parseRequestURL() error {
 	zone := (*qb.parsedProperties)["zone"]
 	port := strconv.Itoa(config.Port)
 	endpoint := config.Protocol + "://" + config.Host + ":" + port
+	if config.Endpoint != "" {
+		endpoint = config.Endpoint
```

Comment:
Maybe we need to parse `endpoint` into `protocol`, `host` and `port` for checking.

Think about following situations:

- endpoint is `192.169.0.10`
- endpoint is `http://192.168.0.10`
- endpoint is `[xxxx:xxxx:xxxx]` (valid ipv6 address)
- endpoint is `https://qingstor.com`
- endpoint is `https://pek3b.qingstor.com`
- endpoint is `https://bucket.pek3b.qingstor.com`
- endpoint is `https://cname-to-qingstor.com`